From 5be727f85e25609fd4cee69c923204066da0f448 Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Mon, 30 Sep 2019 13:24:31 +0100
Subject: [PATCH] OS-7985 platform release gcc4 build fails, breaking smartos
 release page Reviewed by: John Levon <john.levon@joyent.com> Approved by:
 John Levon <john.levon@joyent.com>

---
 tools/build_jenkins | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/tools/build_jenkins b/tools/build_jenkins
index a8ffcefc..080f8fad 100755
--- a/tools/build_jenkins
+++ b/tools/build_jenkins
@@ -173,8 +173,17 @@ case "${PLATFORM_BUILD_FLAVOR}" in
         ;;
 esac
 
-# Update Manta snaplinks for smartos, only if this is a non-debug release build
-if [[ -z "${ENGBLD_CONFIGURE_DEBUG_ARG}" && -n "${ENGBLD_RELEASE}" ]]; then
+#
+# Update Manta snaplinks for smartos, only if this is a non-debug, release
+# build.
+# We check for an empty $PLATFORM_DEBUG_SUFFIX as that gets used as the way to
+# prevent downstream Jenkins builds (e.g. 'platform-gcc4') uploading artifacts
+# to the 'platform' Manta directory, even though they may not strictly be debug
+# builds.
+#
+if [[ -n "${ENGBLD_RELEASE}" && \
+        -z "${ENGBLD_CONFIGURE_DEBUG_ARG}" && \
+        -z "${PLATFORM_DEBUG_SUFFIX}" ]]; then
     case "${PLATFORM_BUILD_FLAVOR}" in
         'smartos'|'triton-and-smartos')
             log_cmd env TRACE=1 ./tools/smartos-release \
-- 
2.21.0

